src/rpc/RPCServer.ts

```diff
@@ -44,10 +43,10 @@ class RPCServer {
   }: {
     manifest: ServerManifest;
     middleware?: MiddlewareFactory<
```

Comment:
Why is `middleware` a `MiddlewareFactory`? I think it should be called `middlewareFactory` since you later call it.

Also please add a docblock to the `middleware` property explaining its types and how it is intended to work.

Reply:
Done.